src/scanspec/specs.py

```diff
@@ -149,6 +162,19 @@ class Product(Spec[Axis]):
     def axes(self) -> list[Axis]:  # noqa: D102
         return self.outer.axes() + self.inner.axes()
 
+    def duration(self):  # noqa: D102
+        if self.outer.duration() is None and self.inner.duration is None:
+            return None
+        else:
+            if self.outer.duration() == self.inner.duration:
+                return self.inner.duration()
+            elif self.outer.duration() is None:
+                return self.inner.duration()
+            elif self.inner.duration():
+                return self.outer.duration()
+            else:
+                return VARIABLE_DURATION
```

Comment:
I think we should say that either the outer or the inner produces duration, not both.

```suggestion
        outer, inner = self.outer.duration(), self.inner.duration()
        if outer is not None and inner is not None:
            raise ValueError("Both outer and inner define a duration")
        return inner if inner is not None else outer
```